### PR TITLE
Add room creation API

### PR DESCRIPTION
**Pull request by rixx**

### CODE REVIEW

**Comment by rashfael** [comment]
Connected the chat creation mockup, everything works!

File: server/venueless/live/modules/world.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    get_world,
+    get_world_config_for_user,
+)
+from venueless.live.exceptions import ConsumerException
+
+logger = logging.getLogger(__name__)
+
+
+class WorldModule:
+    interactive = True
```
Comment (raphaelm) [inline server/venueless/live/modules/world.py:16]: This seems to be unused?

**Comment by rashfael** [comment]
> * We now have room.created and world.updated. Obviously, one approach is more efficient and one is easier to implement. Should we keep both or settle on one?

I like the preservation of intent of `room.create`. It also keeps messages smaller without needing to invent some deep patching syntax.

File: docs/developer/api/world.rst [hunk 2]
```diff
@@ -1,105 +1,37 @@
...
-                        "type": "call.bigbluebutton",
-                        "config": {},
-                        "permissions": []
-                    }
-                ]
-            }
-        ]
+        "name": "Neuer Raum",
+        "modules": [],
+        "announcements": [],
```
[the remark this answers is not in the document]
Comment (rixx) [reply docs/developer/api/world.rst:21]: I'd like to add announcements right away if possible, but we can postpone them, too.

File: server/venueless/live/modules/world.py (hunk 1 above)
Comment (rixx) [reply server/venueless/live/modules/world.py:16]: Yeah, sorry, dividing up five features into three PRs seems to have left artifacts.

**Comment by rixx** [comment]
``world.updated`` (the past tense always trips me up, btw) forces the client to always figure out the diff, which is manageable now, but will get increasingly annoying – plus we don't need to reconstruct the world config every time, and send smaller messages. Not sure the backend simplicity of a single big update call makes up for that?

**Comment by rixx** [comment]
# Announcements

I kind of imagined announcements as just a list of room IDs supplied by the client, and validated by the server (to be discussed: who is allowed to announce where?). They are only sent with the ``room.created`` notification. They'd be shown for a $good time in the UI if one of the rooms in the list is currently open. They'd be transient, so if you join later, you won't see them – they are more intended as a way to say "Hey! You! New room over here!"

**Comment by raphaelm** [comment]
Agreed, let's go for ``room.create``.

> I kind of imagined announcements as just a list of room IDs supplied by the client, and validated by the server (to be discussed: who is allowed to announce where?). They are only sent with the room.created notification. They'd be shown for a $good time in the UI if one of the rooms in the list is currently open. They'd be transient, so if you join later, you won't see them – they are more intended as a way to say "Hey! You! New room over here!"

Ah, hmm, I think room announcements aren't the only announcements we'll have? "The stream is broken, we're working on it, stay tuned" is also an announcement, or do we want to do that separately?

Also, I think it makes sense to "announce" new rooms in the chat timeline as well if there is one, since that's what people's attention often is at, while list of rooms might e.g. even be collapsed on mobile.

**Comment by rixx** [comment]
> Ah, hmm, I think room announcements aren't the only announcements we'll have?

Of course not, no. But "proper" announcements feel like they should persist (at least optionally), and limited to moderators by default, which is an entirely different thing. I imagined room announcements as just transient things, since they are completely bound to that moment in time. Asking people if they'd like to advertise their new room seems like a helpful thing to do.

> Also, I think it makes sense to "announce" new rooms in the chat timeline as well if there is one

Sure, but that's a client-side display issue, isn't it?

**Comment by rixx** [comment]
> I think we should actively reject module types that we don't know. We'll have events that open up the room create permission to everyone and I wouldn't want them to be able to insert their own live streams just because they read our documentation ;)

But rejecting modules we don't know will not stop people from embedding their own streams or BBB instances, because we do know about streams and BBB. If we want to prevent this, we need to find a different restriction or limit room creation to chat rooms for now.

**Pull request merged**